python/ray/autoscaler/updater.py

```diff
@@ -414,17 +416,17 @@ def do_update(self):
             # Run init commands
             self.provider.set_node_tags(
                 self.node_id, {TAG_RAY_NODE_STATUS: STATUS_SETTING_UP})
-            with LogTimer(self.log_prefix +
-                          "Initialization commands completed"):
-                for cmd in self.initialization_commands:
+            with LogTimer(self.log_prefix + "Node setup commands completed"):
+                for cmd in self.node_setup_commands:
                     self.cmd_runner.run(cmd)
 
-            with LogTimer(self.log_prefix + "Setup commands completed"):
-                for cmd in self.setup_commands:
+        if not same_config or self.provider.is_cached(self.node_id):
```

Comment:
Is this correct? Doesn't it mean we will always run these commands every time the node is updated, if it is a cached node?

Reply:
It's intended that `start_commands` should run every time the node is booted (even if it is cached). Is the concern that running e.g. `ray up` will cause this to be rerun? That might be an issue, I'm not sure how to distinguish those two cases.

Reply:
Yeah, since it's rerun on ray up its behaviour is almost identical to the Ray start commands.

Would it actually be reasonable to drop this and use ray start commands for node start commands instead?